### PR TITLE
[Remove]: frontend & bots submodule, improbable-eng/grpc-web

### PR DESCRIPTION
**Pull request by siva2204**
- removed frontend & bots submodule
- removed wrapped improbable-eng/grpc-web server (we'll use envoy here after for grpc-web client)
- add serve mux for rest api
- go fmt
- removed docker files
- news(market events) assets files will be downloaded in public directory and served statically (rest fashion)
- add [air](https://github.com/cosmtrek/air) conf for live reload (not hot reload)

TODO
- dockerize the server with envoy proxy

IMP
-  dalal-street-web will not work without  improbable-eng/grpc-web (can we use it in dev mode ?)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -68,23 +69,16 @@ func RealMain() {
...
-					}
 				} else {
-					resp.WriteHeader(http.StatusBadRequest)
-					resp.Write([]byte("Invalid URL requested"))
+					httpapi.HttpMux.ServeHTTP(resp, req)
 				}
 			},
 		),
 	}
-	models.InspectComponents()
```
Comment (siva2204) [inline main.go:87]: i removed this @abesheknarayan , your thoughts ?

Comment (abesheknarayan) [reply main.go:87]: we can remove it for now and add it later once we have a modified cheat suite

**Comment by abesheknarayan** [comment]
why are docker files removed? Will they be added later on to master branch?

**Comment by abesheknarayan** [comment]
> IMP
> 
> * dalal-street-web will not work without  improbable-eng/grpc-web (can we use it in dev mode ?)

It can be removed as we wont be using dalal-street-web repo anymore

**Comment by abesheknarayan** [comment]
> * removed frontend & bots submodule
> * removed wrapped improbable-eng/grpc-web server (we'll use envoy here after for grpc-web client)
> * add serve mux for rest api
> * go fmt
> * removed docker files
> * news(market events) assets files will be downloaded in public directory and served statically (rest fashion)
> * add [air](https://github.com/cosmtrek/air) conf for live reload (not hot reload)

Everything works as expected!

### Suggestion
Run the httpServer in a seperate goroutine and add graceful shutdown. Graceful shutdown is necessary for gracefully closing out all the hanging resources at the time of stopping the server (Good practice!). More options like `ReadTimeout` and `WriteTimeout` can also be set in httpServer config (need to think about what is the ideal time for doing that).

For more ref: [link](https://github.com/abesheknarayan/microservice-go/blob/8cfc996578bb3f09fba0ff8f88a49d05fefe3eb1/main.go#L14)

**Comment by abesheknarayan** [comment]
### Remove vulnerable dependency

`github.com/dgrijalva/jwt-go` is deprecated, use `github.com/golang-jwt/jwt`

**Comment by siva2204** [comment]
- migrated to `github.com/golang-jwt/jwt`

## TODO 

- graceful http server shutdown
- dockerize

these todos can be done later

**Review by abesheknarayan — APPROVED** [review]

**Pull request merged**